core/lib/spree/core/controller_helpers/auth.rb

```diff
@@ -54,6 +54,11 @@ def store_location
           Spree::UserLastUrlStorer.new(self).store_location
         end
 
+        # Auth extensions are expected to define it, otherwise it's a no-op
+        def spree_current_user
+          defined?(super) ? super : nil
+        end
```

Comment:
Some lines later, in `try_spree_current_user`, we have:

```ruby
        def try_spree_current_user
          # This one will be defined by apps looking to hook into Spree
          # As per authentication_helpers.rb
          if respond_to?(:spree_current_user, true)
            spree_current_user
          # This one will be defined by Devise
          elsif respond_to?(:current_spree_user, true)
            current_spree_user
          end
        end```

With the proposed change, will the `respond_to?(:spree_current_user, true)` always be true?

Reply:
Yes, but I think that's ok. Then `spree_current_user` will delegate to the auth library, and `solidus_auth_devise` is already [implementing](https://github.com/solidusio/solidus_auth_devise/commit/69269765) it as noted in the linked PR.

My understanding is that `#current_spree_user` is never defined (`solidus_auth_devise` delegates to `#current_user` instead), so it's safe to forgo its support.

Reply:
I'm not sure to fully understand here. If we are 100% sure that the `elsif` will never be executed, why keep it?

Reply:
Yes, I guess we can remove it. Pinging @elia to be 100% sure.

Reply:
@waiting-for-dev you're right, we shouldn't care about `current_spree_user` and this will work.
I'm just a bit hesitant on having a cover for a missing authentication implementation. Thoughts on having a sort of `NoAuth` module that we inject from within an initializer like we do for `Spree::AuthenticationHelpers` in auth-devise?

_solidus_auth_devise/lib/spree/auth/engine.rb_

```rb
module Spree
  module Auth
    class Engine < Rails::Engine
      …
      config.to_prepare do
        Spree::Auth::Engine.prepare_backend if SolidusSupport.backend_available?
        Spree::Auth::Engine.prepare_frontend if SolidusSupport.frontend_available?

        ApplicationController.include Spree::AuthenticationHelpers
      end
```

in the no-auth case would be something like:

_my-app/config/initializers/solidus_auth.rb_

```rb
Rails.application.reloader.to_prepare do
  ApplicationController.include Spree::NoAuth
end
```

Reply:
@elia given our new storefront customization strategy is never to inject anything, but rather to copy-paste code into the host application. We don't want to do anything weird with the storefront as that makes customization harder.

Reply:
Thanks for looking into it, @elia! On top of what @aldesantis said, I think it's not worth it to make things complex. I agree that we don't have a clear separation of concerns for the authentication system. But extracting things to separated modules would give an *apparent* impression of orthogonality, but the truth is that all the components expect something to be there. Until we don't revisit it, I think it's better to have the no-op behavior as visible as possible.